### PR TITLE
Added huber_loss

### PR DESCRIPTION
**Pull request by ZiadAmerr**
Closes https://github.com/unifyai/ivy/issues/7065

### CODE REVIEW

**Comment by kurshakuz** [comment]
Hey @ZiadAmerr, thank you for your conrtribution! It is looking good, but can you please show the result of the test run locally for your implemented method?

Thanks!

**Review by kurshakuz — COMMENTED** [review]
Hey @ZiadAmerr, thank you for your conrtribution! It is looking good, but can you please show the result of the test run locally for your implemented method?

Thanks!

**Comment by ZiadAmerr** [comment]
Hi @kurshakuz. I am having an issue with running the test on my PC. PyCharm on my mac doesnt allow me to install TensorFlow, it is a problem related to all MacBook M1's. I have been trying to fix it for 3 days now and I couldn't. I will try to provide the output of the tests, but is the code ok?

**Comment by kurshakuz** [comment]
Hey! Were you able to run tests?

**Comment by ZiadAmerr** [comment]
Hi @kurshakuz, nope, not yet, I have sent my error on Discord channel and I have been trying to fix it but I just couldn't, is there any way we can contact in real time, maybe you can help me bypass it?

**Comment by ZiadAmerr** [comment]
#7065

**Comment by kurshakuz** [comment]
Hey! Can you send the link to your message in discord so I can have a look and respond you there as well? Thanks

**Comment by ZiadAmerr** [comment]
https://discord.com/channels/@me/1045239504400818207/1045239534067130439

**Comment by kurshakuz** [comment]
Hey @ZiadAmerr!

I have checked your PR and noticed a couple of things that needs to be fixed. It currently throws following errors that need to be resolved.

Please firstly fix following two errors in your test:
```
E   hypothesis.errors.InvalidArgument: test_torch_huber_loss() got an unexpected keyword argument 'size_average', from `size_average=booleans()` in @given
```
```
E   hypothesis.errors.InvalidArgument: test_torch_huber_loss() got an unexpected keyword argument 'reduce', from `reduce=booleans()` in @given
```

After fixing those,  make sure that your method return correct data types as it will then show the following error:
```
E       AssertionError: the return with a torch backend produced data type of float16, while the return with a torch backend returned a data type of float32.
```

After fixing those, please feel free to request a new review 🙂

**Review by ZiadAmerr — COMMENTED** [review]
I think I have fixed the problems related to the testing code. Can you please test it and tell me if there is any problems with it as my testing problem is still there and I still can't run the `pytest`.

But I couldn't find the reason behind the following error:
`E       AssertionError: the return with a torch backend produced data type of float16, while the return with a torch backend returned a data type of float32.`

**Review by ZiadAmerr — COMMENTED** [review]
Minor change

**Review by ZiadAmerr — COMMENTED** [review]
Updated some files that might have been causing the linting issue

**Comment by kurshakuz** [comment]
Hey @ZiadAmerr ! Sure, I can the test locally, but can you first resolve the conflicts and fix the linting error? Thanks 🙂

File: ivy_tests/test_ivy/test_frontends/test_torch/test_loss_functions.py [hunk 1]
```diff
@@ -269,3 +269,66 @@ def test_torch_l1_loss(
...
+):
+    pred_dtype, pred = dtype_and_pred  # Input
+    true_dtype, true = dtype_and_true  # Target
+    helpers.test_frontend_function(
+        input_dtypes=[pred_dtype[0], true_dtype[0]],
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        frontend=frontend,
```
Comment (kurshakuz) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_loss_functions.py:328]: this fronted variable is not defined in the scope right now. Can you fix it as well?

**Review by ZiadAmerr — COMMENTED** [review]
I have finalized all errors but I am still running into this `AssertionError`, I have been working on it for 5 days now and I really can't identify the reason behind it. I even tried using the testing code thats available from other functions like `mse_loss` and `l1_loss` and modifying it to test my `huber_loss` but it raises the same error. Also I think the testing function for `smooth_l1_loss` is missing

By the way, I can run the tests locally now, thanks a ton for your previous help

**Comment by kurshakuz** [comment]
Thanks for noticing the missing test!

With regards to your issues with data types, try playing around with `with_unsupported_dtypes` as this may help. You can read more about it here: https://lets-unify.ai/ivy/deep_dive/data_types.html?highlight=with_unsupported_dtypes#supported-and-unsupported-data-types

**Comment by ZiadAmerr** [comment]
Well I came across that but I decided not to use it because why would I restrict my function to specific data types.

In other words, if torch's `huber_loss` does support `float16`, why would I restrict it? Thats why I didn't use that decorator

**Review by ZiadAmerr — COMMENTED** [review]
All done, added `huber_loss`, `test_huber_loss`, `test_smooth_l1_loss`. All have been tested and working locally

**Pull request closed without merging**